tests/Test/Xrefcheck/TooManyRequestsSpec.hs

```diff
@@ -149,35 +158,118 @@ test_tooManyRequests = testGroup "429 response tests"
                     )
                 | otherwise -> toResponse ("" :: Text, serviceUnavailable503)
       infoReverseAccumulatorRef <- newIORef []
+      setRef <- newIORef S.empty
       E.bracket (forkIO $ mock429WithGlobalIORef infoReverseAccumulatorRef) killThread $ \_ -> do
-        _ <- verifyLink "http://127.0.0.1:5000/429grandfinale"
+        _ <- verifyLinkDefault setRef "http://127.0.0.1:5000/429grandfinale"
         infoReverseAccumulator <- readIORef infoReverseAccumulatorRef
         reverse infoReverseAccumulator @?=
           [ ("HEAD", tooManyRequests429)
           , ("HEAD", serviceUnavailable503)
           , ("GET", ok200)
           ]
+    , testGroup "Timeout tests"
+      [ testCase "Returns an error on timeout if there were no 429 responses" $
+          timeoutTestCase [Delay] False
+            (cNetworkingL . ncMaxTimeoutRetriesL .~ 2)
+      , testCase "Returns an error if there were 429 but no timeouts allowed" $
+          timeoutTestCase [Respond429, Delay] False
+            (cNetworkingL . ncMaxTimeoutRetriesL .~ 0)
+      , testCase "Succeeds if there were 429 and one timeout allowed" $
+          timeoutTestCase [Respond429, Delay] True
+            (cNetworkingL . ncMaxTimeoutRetriesL .~ 1)
+      , testCase "Fails on second timeout if there were 429 and one timeout allowed" $
+          timeoutTestCase [Respond429, Delay, Delay] False
+            (cNetworkingL . ncMaxTimeoutRetriesL .~ 1)
+      , testCase "Fails on second timeout if there were 429 and one timeout allowed" $
+          timeoutTestCase [Respond429, Delay, Delay] False
+            (cNetworkingL . ncMaxTimeoutRetriesL .~ 1)
+      , testCase "Fails on maximum allowed errors achieved (mixed errors)" $
+          timeoutTestCase [Respond429, Delay, Respond429, Delay] False $ \c -> c
+            & cNetworkingL . ncMaxTimeoutRetriesL .~ 3
+            & cNetworkingL . ncMaxRetriesL .~ 3
+      , testCase "Fails on timeout if another domain returned 429" $ do
+          setRef <- newIORef S.empty
+          checkMultipleLinksWithServer
+            (mockTimeout (sec 0.4) [Respond429, Ok, Delay])
+            setRef
+            [ VerifyLinkTestEntry{ vlteConfigModifier = \c -> c
+                                     & cNetworkingL . ncMaxRetriesL .~ 1
+                                     & setAllowedTimeout
+                                 , vlteLink = "http://127.0.0.1:5000/timeout"
+                                 , vlteProgress = mkProgressWithOneTask True
+                                 , vlteExpectation = VerifyResult []
+                                 }
+            , VerifyLinkTestEntry{ vlteConfigModifier = \c -> c
+                                     & cNetworkingL . ncMaxTimeoutRetriesL .~ 0
+                                     & setAllowedTimeout
+                                 , vlteLink = "http://localhost:5000/timeout"
```

Comment:
`127.0.0.1:5000` vs `localhost:5000`? That's smart :trollface: 

I guess this test counts, cannot imagine what can practically go wrong.